libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp

```diff
@@ -86,7 +86,7 @@ AP_BattMonitor_Analog::AP_BattMonitor_Analog(AP_BattMonitor &mon,
     }
 #endif
     _state.var_info = var_info;
-    
+    //AR
```

Comment:
unneeded whitespace change

Reply:
//AR is sort of bookmark indicating problematic block of code (was debugging ADC ), but you are right, it souldn't be commited to the main repo.

Reply:
My remarks for the points:

1. I agree, I will fix it.

2. I've added the ESP_PLATFORM define in the code branch for ESP32 targets. This works around a bug in ESP-IDF v5.3, which doesn’t define ESP_PLATFORM automatically and breaks some FreeRTOS code.

3. It’s a minor bug. I just noticed it on WSL and fixed it.

4. I tested the build system and install scripts on a clean Ubuntu 24 installation (for both ESP32 and ESP32S3 targets) and tested the built firmware on the ESP32S3 target using the esp32s3devkit.h board definition.
Then the list of fixes:
- Fixed all GCC 13 compatibility issues everywhere.
- Fixed the RCIn issue (it was broken on S3).
- Added enhancements to the ADC (ported to the new driver).
- Fixed WDT issues with ESP-IDF v5.3.
- Added support for the IBUS hack for the FlySky IA6 receiver.

5. I noticed some printf statements that I believe should only be present in debug builds, so I fixed them during debugging. It's not a big deal.